fastify.d.ts

```diff
@@ -301,8 +301,8 @@ declare namespace fastify {
   /**
    * Register options
    */
-  interface RegisterOptions<HttpServer, HttpRequest, HttpResponse> {
-    [key: string]: any,
+   type RegisterOptions<HttpServer, HttpRequest, HttpResponse, T extends {}> = {
```

Comment:
Why `extends {}`? Is `T` supposed to be an `Object`? If so I'd recommend changing this to `T extends Object`